lib/src/isolate_manager/isolate_manager.dart

```diff
@@ -0,0 +1,69 @@
+import 'dart:async';
+
+import 'package:dash_kit_network/dash_kit_network.dart';
+
+// This is an abstract class for managing isolates It defines the basic methods
+// needed to start, stop and send messages to an isolate. Flutter Web doesn't
+// support isolates so the `IsolateManager` class has a different implementation
+// for web.
+// ignore: prefer-match-file-name
+abstract class IsolateManagerInterface {
+  // Method for starting the isolate. Should be called before sending messages.
+  Future<void> start();
+
+  // Method for stopping the isolate. Should be called when the isolate is no
+  //longer needed.
+  void stop();
+
+  // Method for sending messages to the isolate. The `mapper` function is
+  // executed in the isolate and the result is returned.
+  Future<FutureOr<T>> send<T>(
```

Comment:
Also if it is possible let's take into account the best practices for code documentation, for example from here: https://medium.com/codex/flutter-dart-documentation-791371ff2e0f